Coursera/Arrays,Lists and Structured data/Week3UniqueIPLogAnalyzer.java

```diff
@@ -0,0 +1,198 @@
+package Assignments2;
+
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import edu.duke.*;
+
+public class Week3UniqueIPLogAnalyzer
+{
+    private ArrayList<Week3UniqueIPLogEntry> records;
+
+    public Week3UniqueIPLogAnalyzer() {
+        records = new ArrayList<Week3UniqueIPLogEntry>();
+    }
+
+    public void readFile(String filename) {
+        FileResource fr = new FileResource(filename);
+
+        for (String line : fr.lines()) {
+            Week3UniqueIPWebLogParser parser = new Week3UniqueIPWebLogParser();
+            Week3UniqueIPLogEntry newEntry = parser.parseEntry(line);
+            records.add(newEntry);
+        }
+    }
+
+    public void printAll() {
+        for (Week3UniqueIPLogEntry le : records) {
+            System.out.println(le);
+        }
+    }
+
+    public int countUniqueIPs() {
+        ArrayList<String> uniqueIps = new ArrayList<String>();
+
+        for (Week3UniqueIPLogEntry le : records) {
+            if (!uniqueIps.contains(le.getIpAddress())) {
+                uniqueIps.add(le.getIpAddress());
+            }
+        }
+
+        return uniqueIps.size();
+    }
+
+    public void printAllHigherThanNum(int num) {
+        for (Week3UniqueIPLogEntry le : records) {
+            if (le.getStatusCode() > num) {
+                System.out.println(le);
+            }
+        }
+    }
+
+    public ArrayList<String> uniqueIPVisitsOnDay(String someday) {
+        ArrayList<String> uniqueIps = new ArrayList<String>();
+
+        for (Week3UniqueIPLogEntry le : records) {
+            if (le.getAccessTime().toString().contains(someday)) {
+                if (!uniqueIps.contains(le.getIpAddress())) {
+                    uniqueIps.add(le.getIpAddress());
+                }
+            }
+        }
+
+        return uniqueIps;
+    }
+
+    public int countUniqueIPsInRange(int low, int high) {
+        ArrayList<String> uniqueIps = new ArrayList<String>();
+
+        for (Week3UniqueIPLogEntry le : records) {
+            if (le.getStatusCode() >= low && le.getStatusCode() <= high) {
+                if (!uniqueIps.contains(le.getIpAddress())) {
+                    uniqueIps.add(le.getIpAddress());
+                }
+            }
+        }
+
+        return uniqueIps.size();
+    }
+}
+
+public class Week3UniqueIPLogEntry
```

Comment:
You can put week names as folder names and let classes be defined as it is in those folders.

Reply:
I kept them for easy identification in intelliJ to store them all in one single folder "coursera".